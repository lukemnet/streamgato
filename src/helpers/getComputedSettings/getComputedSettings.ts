import { ShorthandValues } from 'types';

const getComputedSettings = (settings: ShorthandValues, defaultSettings: ShorthandValues) => {
  const settingKeys = Object.keys(settings);
  let computedSettings = {} as ShorthandValues;

  settingKeys.forEach((settingKey) => {
    const value = settings[settingKey];
    const defaultValue = defaultSettings[settingKey];

    if (value && value !== defaultValue) {
      computedSettings[settingKey] = value;
    }
  });

  return computedSettings;
}

export default getComputedSettings;
