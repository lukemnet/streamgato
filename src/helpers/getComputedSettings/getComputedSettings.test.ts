import getComputedSettings from './getComputedSettings';
import { ShorthandValues } from 'types';

const testSettings = {
  h: 1,
  m: 2,
  s: 3,
} as ShorthandValues;

const defaultSettings = {
  h: 2,
  m: 2,
  s: 3,
} as ShorthandValues;

it('should compute settings correctly', () => {
  expect(getComputedSettings(testSettings, defaultSettings)).toMatchSnapshot();
})