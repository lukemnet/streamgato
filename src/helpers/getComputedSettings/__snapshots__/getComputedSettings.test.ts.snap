// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should compute settings correctly 1`] = `
Object {
  "h": 1,
}
`;
